### PR TITLE
DOC/REL: v0.14.0

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Add release notes for v0.14.0
- Fix various malformed headers in the release notes page, which brings us from some wild malformed docs site to something more reasonable.
- Fix miscellaneous other errors such as non-links and bullet point formatting (the ones I noticed when browsing the docs build, probably not everything).

## Motivation and Context

I want to use the new filter in lucid and there are many places where we'll want to use `WeakPartialSlot`

## How Has This Been Tested?

I checked the built docs

## Where Has This Been Documented?

It is docs

## Screenshots:
before:
image

after:
image

### CODE REVIEW

**Comment by ZLLentz** [comment]
I thought this was done, but actually the in-docs links don't work because the new classes don't show up in the API pages (?)
Investigating a little bit

**Comment by ZLLentz** [comment]
There was a bug in the api list creator where it was ignoring all classes in the api because they weren't subclasses of "empty tuple". I switched this to subclasses of "object" and it behaves much better.

File: docs/source/releases.rst [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+- zllentz
 
 
 v0.13.0 (2023-09-27)
 ====================
 
 Features
 --------
-- Add cleaned up and performant version of heavily-used ``get_info`` in
+- Add cleaned up and performant version of heavily-used :any:`get_info` in
```
Comment (tangkong) [inline docs/source/releases.rst:25]: The `:any:` cross-reference is new to me, it's a nice fallback if we don't want to specify `:class:` or something.

File: docs/update_api_list.py [hunk 2]
```diff
@@ -117,7 +117,7 @@ def sort_key(obj):
 ]
 
 classes = find_all_classes(
-    (),
+    (object),
```
Comment (tangkong) [inline docs/update_api_list.py:120]: Hah so we just weren't collecting any classes ever. This is a part of a documentation-building cookiecutter/skeleton/tooling that isn't super uniform, should we be adding this to all our "Python Library" packages?  🤔

**Review by tangkong — APPROVED** [review]
LGTM, good catch on the docs.

File: docs/source/releases.rst (hunk 1 above)
Comment (ZLLentz) [reply docs/source/releases.rst:25]: Not only does it save you from specifying, it saves you from getting the whole path to the object, so you don't need to do
```
:class:`pcdsutils.modulename.classname`
```

File: docs/update_api_list.py (hunk 2 above)
Comment (ZLLentz) [reply docs/update_api_list.py:120]: I'm wondering if this belongs as a standard tool in some form in e.g. pcdsutils or in a standalone module that we can pull in as a docs dependency.

Comment (ZLLentz) [reply docs/update_api_list.py:120]: It's certainly a useful way to generate a full project API that is easier to browse than any more straightforward method in sphinx

**Pull request merged**